discovery-provider/src/tasks/index.py

```diff
@@ -363,306 +504,106 @@ def index_blocks(self, db, blocks_list):
+                for tx in sorted_txs:
+                    tx_hash = web3.toHex(tx["hash"])
+                    tx_target_contract_address = tx["to"]
+                    tx_receipt = tx_receipt_dict[tx_hash]
+                    should_skip_tx = (tx_target_contract_address == zero_address) or (
+                        skip_tx_hash is not None and skip_tx_hash == tx_hash
                     )
-                    user_factory_txs.append(tx_receipt)
 
-                # Handle track operations
-                if (
-                    tx_target_contract_address
-                    == get_contract_addresses()["track_factory"]
-                ):
-                    logger.info(
-                        f"index.py | TrackFactory contract addr: {tx_target_contract_address}"
-                        f" tx from block - {tx}, receipt - {tx_receipt}"
-                    )
-                    # Track state operations
-                    track_factory_txs.append(tx_receipt)
-
-                # Handle social operations
-                if (
-                    tx_target_contract_address
-                    == get_contract_addresses()["social_feature_factory"]
-                ):
-                    logger.info(
-                        f"index.py | Social feature contract addr: {tx_target_contract_address}"
-                        f"tx from block - {tx}, receipt - {tx_receipt}"
+                    if should_skip_tx:
+                        logger.info(
+                            f"index.py | Skipping tx {tx_hash} targeting {tx_target_contract_address}"
+                        )
+                        continue
+                    else:
+                        group_transaction_by_type(TXS_GROUPED_BY_TYPE, tx, tx_receipt)
+
+                # pre-fetch cids asynchronously to not have it block in user_state_update
+                # and track_state_update
+                try:
+                    ipfs_metadata, blacklisted_cids = fetch_ipfs_metadata(
+                        db,
+                        TXS_GROUPED_BY_TYPE[USER_FACTORY],
+                        TXS_GROUPED_BY_TYPE[TRACK_FACTORY],
+                        block_number,
+                        block_hash,
                     )
-                    social_feature_factory_txs.append(tx_receipt)
 
-                # Handle repost operations
-                if (
-                    tx_target_contract_address
-                    == get_contract_addresses()["playlist_factory"]
-                ):
-                    logger.info(
-                        f"index.py | Playlist contract addr: {tx_target_contract_address}"
-                        f"tx from block - {tx}, receipt - {tx_receipt}"
+                    # bulk process operations once all tx's for block have been parsed
+                    # and get changed entity IDs for cache clearing
+                    # after session commit
+                    changed_entity_ids_map = bulk_process_state_changes(
```

Comment:
we delegate the handling of tx processing operations to this helper function

Reply:
love it, but maybe just `process_state_changes`? bulk to me suggests a bit more "bulk" than what's going on here where we're just parsing & then writing to the db